sound/soc/sof/sof-audio.c

```diff
@@ -97,8 +106,8 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (!swidget->private)
 		return 0;
 
-	/* only free when refcount is 0 */
-	if (--swidget->use_count)
+	/* only free when use_count is 0 */
+	if (--swidget->use_count > 0)
```

Comment:
this looks like it should be squashed earlier, and I also don't get what the > 0 brings?